lib/tools.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+/*
+ * A bunch of handy functions that should probably live in a module somewhere.
+ */
+
+function filterByKey(object, filter) {
+    var filtered = {};
+    Object.keys(object).forEach(function (key) {
+        if (filter(key)) {
+            filtered[key] = object[key];
+        }
+    });
+    return filtered;
+}
+
+function find(xs, predicate) {
```

Comment:
AFAIK, Array.find() is part of Harmony, and is provided by [es6-shim](https://github.com/paulmillr/es6-shim#safe-shims), with the difference being it returns the value, not the index in the array, so maybe we should use that one?